x-pack/platform/plugins/private/reporting/server/lib/content_stream.ts

```diff
@@ -319,3 +321,19 @@ export async function getContentStream(
     parameters
   );
 }
+
+export async function finishedWithNoPendingCallbacks(stream: Writable) {
+  await finished(stream, { readable: false });
+
+  // Race condition workaround:
+  // `finished(...)` will resolve while there's still pending callbacks in the writable part of the `stream`.
+  // This introduces a race condition where the code continues before the writable part has completely finished.
+  // The `pendingCallbacks` function is a hack to ensure that all pending callbacks have been called before continuing.
+  // For more information, see: https://github.com/nodejs/node/issues/46170
+  await (async function pendingCallbacks(delay = 1) {
```

Comment:
We recently got pinged for a potential infinite recursion case, this looks like potentially another.  Can we put an upper time limit on this?